### PR TITLE
Rdyn

### PR DESCRIPTION
**Pull request by Dyndyn**

### CODE REVIEW

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Created by dyndyn on 05.02.2017.
+ */
+@Controller
+@RequestMapping("/storage")
+public class StorageController {
+
+    @GetMapping
+    public ModelAndView getPage(@RequestParam("user_id") int userId) {
+        ModelAndView model = new ModelAndView("storage");
+        final String uri = "http://localhost:8080/rest/storage/" + userId;
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java]: Move this to property file. I have already written the same comment for Vitaliy's PR.

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<Storage[]> storages = restTemplate.getForEntity(uri, Storage[].class);
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java]: Better to get list right here instead of transforming array to list below

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        model.addObject("list", Arrays.asList(storages.getBody()));
+        return model;
+
+    }
+
+    @PostMapping("/update")
+    @ResponseStatus(value = HttpStatus.OK)
+    public void updateAmount(@RequestParam("userId") int userId, @RequestParam("productId") int productId, @RequestParam("amount") int amount) {
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java]: "userId", "productId" and "amount" is optional here. Yo cam omit them

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        Storage storage = new Storage(new User(), new Product(), amount, null);
+        storage.getUser().setId(userId);
+        storage.getProduct().setId(productId);
+
+        final String uri = "http://localhost:8080/rest/storage/";
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorageController.java]: to property file

File: MVCapplication/src/main/webapp/WEB-INF/views/storage.jsp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                            <p class="fa fa-minus"></p></td>
+                    </tr>
+                </c:forEach>
+                </tbody>
+            </table>
+        </div>
+    </div>
+</div>
+
+<script type="text/javascript" language="JavaScript">
```
Comment (kucherif) [inline MVCapplication/src/main/webapp/WEB-INF/views/storage.jsp]: Java Script should be in the separate file.

File: MVCapplication/src/main/webapp/WEB-INF/views/storage.jsp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    function minus(userId, productId, amount, index) {
+        amount--;
+        if (amount >= 0) {
+            $.ajax({
+                url: "http://localhost:8080/client/storage/update",
```
Comment (kucherif) [inline MVCapplication/src/main/webapp/WEB-INF/views/storage.jsp]: ??? use relative path

File: RESTserver/src/main/java/com/softserve/if072/restservice/controller/StorageController.java [hunk 7]
```diff
@@ -64,38 +64,23 @@ public void delete(@PathVariable int id, HttpServletResponse response) {
...
     @ResponseStatus(value = HttpStatus.OK)
     public void update(@RequestBody Storage storage, HttpServletResponse response) {
-        int id = storage.getId();
         try {
             storageService.update(storage);
-            LOGGER.info(String.format("Store with id %d was updated", id));
+            LOGGER.info("Store was updated");
         } catch (DataNotFoundException e) {
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
-            LOGGER.error(String.format("Cannot update Storage with id %d", id), e);
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/controller/StorageController.java]: why did you remove useful information from the logs? 
This is related to this row and above rows

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java [hunk 8]
```diff
@@ -4,34 +4,29 @@
...
     @Autowired
-    private ShoppingListDAO shoppingListDAO;
+    public ShoppingListService(ShoppingListDAO shoppingListDAO) {
+        this.shoppingListDAO = shoppingListDAO;
+    }
 
     public List<ShoppingList> getByUserId(int user_id) throws DataNotFoundException {
         List<ShoppingList> list = shoppingListDAO.getByUserID(user_id);
-        if (list != null && !list.isEmpty()){
+        if (list != null && !list.isEmpty()) {
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java]: Please ask guys about CollectionUtils.isNotEmpty()

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java [hunk 9]
```diff
@@ -40,12 +35,11 @@ public void update(ShoppingList shoppingList) throws DataNotFoundException {
...
-    public void delete(int id) throws DataNotFoundException {
-        ShoppingList shoppingList = shoppingListDAO.getByID(id);
-        if (shoppingList != null){
-            shoppingListDAO.deleteById(id);
+    public void delete(ShoppingList shoppingList) throws DataNotFoundException {
+        if (shoppingList != null) {
+            shoppingListDAO.delete(shoppingList);
         } else {
-            throw new DataNotFoundException(String.format("ShoppingList with id %d was not found", id));
+            throw new DataNotFoundException("ShoppingList was not found");
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java]: why did you remove useful information from the log?

Comment (Dyndyn) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java]: because ShoppingList doesn't have id.

Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java]: But it has some other useful fields, by which it can be identified, for example user id, maybe something else.
